src/binary/mod.rs

```diff
@@ -246,6 +247,13 @@ where
         None
     };
 
+    let kernel_info = KernelInfo {
+        kernel_base: kernel_bytes.as_ptr() as u64,
```

Comment:
This is the physical memory address of the kernel ELF file. To get the virtual address at which the kernel is mapped, we need to look at the segments in the program header of the ELF file. Note that there is no guarantee that these segments are continuous in memory since each section can be mapped arbitrarily using a linker script.